torch/_refs/__init__.py

```diff
@@ -3803,7 +3808,10 @@ def _reshape_view_helper(a: TensorLikeType, *shape, allow_copy: bool) -> TensorL
         # specify the same number of elements above
         accum = a_.shape[idx]
         end = idx
-        while guard_size_oblivious(accum % length != 0):
+        while not guard_or_false(accum % length == 0):
```

Comment:
For this I was tbh thinking the following:
run the this loop and all code after line 3790 with out any guard_or_false
or guard_size_obliv (maybe only statically_known_true) and have a large try catch for data dependent error 
that would fall back to the as_strided approach